packages/ndla-ui/src/Resource/Resource.tsx

```diff
@@ -0,0 +1,316 @@
+  min-width: ${(p) => (p.imageSize === 'normal' ? '136px' : '56px')};
+  height: ${(p) => (p.imageSize === 'normal' ? '96px' : '40px')};
+`;
+
+export interface ListResourceProps {
+  link: string;
+  title: string;
+  resourceImage: ResourceImageProps;
+  topics: string[];
+  tags?: string[];
+  description?: string;
+  actionMenu?: ReactNode;
+}
+
+export const ListResource = ({
+  link,
+  title,
+  tags,
+  resourceImage,
+  topics,
+  description,
+  actionMenu,
+}: ListResourceProps) => {
+  const showDescription = description !== undefined;
+  return (
+    <ResourceWrapper to={link}>
+      <StyledImage alt={resourceImage.alt} src={resourceImage.src} imageSize={showDescription ? 'normal' : 'compact'} />
+      <ResourceInfoWrapper>
+        <Row>
+          <ResourceTitle>{title}</ResourceTitle>
+          <TagList tags={tags} />
+          {actionMenu}
+        </Row>
+        <Row>
+          <TopicList topics={topics} />
+        </Row>
+        {showDescription && (
+          <Row>
+            <ResourceDescription>{description}</ResourceDescription>
+          </Row>
+        )}
+      </ResourceInfoWrapper>
+    </ResourceWrapper>
+  );
+};
+
+interface BlockElementWrapperProps {
+  maxWidth: string;
+  maxHeight: string;
+}
+
+const BlockDescription = styled.p`
+  display: -webkit-box;
+  line-clamp: 2;
+  ${fonts.sizes(16)};
+  height: 0em;
+  margin: 0;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  transition: height 0.2s ease-out;
+`;
+
+const BlockElementWrapper = styled(SafeLink)<BlockElementWrapperProps>`
+  display: flex;
+  text-decoration: none;
+  box-shadow: none;
+  flex-direction: column;
+  max-width: ${(p) => p.maxWidth};
+  max-height: ${(p) => p.maxHeight};
+  font-family: ${fonts.sans};
+  border: 1px solid ${colors.brand.light};
+  border-radius: 2px;
+  color: ${colors.brand.greyDark};
+  &:hover {
+    p {
+      // babel throws an error when BlockDescription is selected directly.
+      // Unfortunate css needed for multi-line text overflow ellipsis.
+      height: 3.1em;
+      -webkit-line-clamp: 2;
+      line-clamp: 2;
+      -webkit-box-orient: vertical;
+    }
+  }
+`;
+
+const RightRow = styled(Row)`
+  justify-content: flex-end;
+`;
+
+const BlockInfoWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  padding: ${spacing.small};
+  gap: ${spacing.xxsmall};
+`;
+
+interface ImageWrapperProps {
+  maxWidth: string;
+}
+
+const ImageWrapper = styled.div<ImageWrapperProps>`
+  display: flex;
+  max-width: ${(p) => p.maxWidth};
+  width: 100%;
+  overflow: hidden;
+  align-items: center;
+  div {
+    min-width: 100%;
+  }
+  img {
+    min-width: 100%;
+  }
+`;
+
+interface BlockResourceProps {
+  link: string;
+  title: string;
+  resourceImage: ResourceImageProps;
+  topics: string[];
+  tags?: string[];
+  description?: string;
+  maxWidth?: string;
+  maxHeight?: string;
+  actionMenu?: ReactNode;
+}
+
+export const BlockResource = ({
```

Comment:
Kan splitte ut BlockResource og ListResource til egne filer for å redusere lengden på filen.

Reply:
Valgte å ha det i samme fil da `BlockResource` tok i bruk flere av komponentene jeg lagde for `ListResource`. Skal jeg splitt det i tre filer, der jeg heller har en delt komponent-fil?